src/transport/SessionHandle.h

```diff
@@ -70,6 +75,7 @@ class SessionHandle
     NodeId GetPeerNodeId() const { return mPeerNodeId; }
     const Optional<uint16_t> & GetPeerSessionId() const { return mPeerSessionId; }
     const Optional<uint16_t> & GetLocalSessionId() const { return mLocalSessionId; }
+    const Optional<GroupId> & GetGroupId() const { return mGroupId; }
```

Comment:
This is going to collide with https://github.com/project-chip/connectedhomeip/pull/11266

We presumably want a third type in the variant once both PRs are in.  Please talk to @kghost about how to best coordinate this.